kinesis/kinesis.go

```diff
@@ -562,15 +562,32 @@ func (outputPlugin *OutputPlugin) randomString() string {
 	return string(outputPlugin.random.buffer)
 }
 
+func getFromMap(dataKey string, records map[interface{}]interface{}) interface{} {
+	for k, v := range records {
+		currentKey := stringOrByteArray(k)
+		if currentKey == dataKey {
+			return v
+		}
+	}
+
+	return nil
+}
+
 // getPartitionKey returns the value for a given valid key
 // if the given key is empty or invalid, it returns a random string
 func (outputPlugin *OutputPlugin) getPartitionKey(record map[interface{}]interface{}) string {
 	partitionKey := outputPlugin.partitionKey
+	partitionKeys := strings.Split(partitionKey, "->")
+	num := len(partitionKeys)
 	if partitionKey != "" {
-		for k, v := range record {
-			dataKey := stringOrByteArray(k)
-			if dataKey == partitionKey {
-				value := stringOrByteArray(v)
+		for count, dataKey := range partitionKeys {
+			newRecord := getFromMap(dataKey, record)
+			_, ok := newRecord.(map[interface{}]interface{})
+			if ok {
+				record = newRecord.(map[interface{}]interface{})
+			} 
```

Comment:
I don't think this code properly accounts for error cases... please test those

What happens if the newRecord can not be cast to a map?

I think in that case you need break out of the loop

Reply:
yeah.. make sense. I lost this part after I updated the code. Will add it in next version